src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemDatasetValidator.Cancel.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Globalization;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Core.Features.Workitem.Model;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    public sealed class CancelWorkitemDatasetValidator : WorkitemDatasetValidator
+    {
+        protected override void OnValidate(DicomDataset dataset)
+        {
+            // all the attributes in the dataset are optional. Hence no validation is being done here.
```

Comment:
We should check against the workitem from blob here, and we should check that Final State requirements are being met, as in https://dicom.nema.org/medical/dicom/current/output/html/part04.html#sect_CC.2.5-1.

We can use a similar pattern to the requirement validation in the Add validator

Reply:
Done